### PR TITLE
upgrade to selenium-java 4.0.0-alpha-2

### PR DESCRIPTION
**Pull request by asolntsev**
## Proposed changes
Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.
If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Checklist
- [ ] Checkstyle and unit tests pass locally with my changes by running `gradle check chrome htmlunit` command
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/main/java/com/codeborne/selenide/CollectionCondition.java [hunk 1]
```diff
@@ -11,9 +11,9 @@
 import com.codeborne.selenide.collections.TextsInAnyOrder;
 import com.codeborne.selenide.impl.WebElementsCollection;
 import com.google.common.base.Predicate;
-import org.checkerframework.checker.nullness.compatqual.NullableDecl;
 import org.openqa.selenium.WebElement;
 
+import javax.annotation.Nullable;
```
Comment (rosolko) [inline src/main/java/com/codeborne/selenide/CollectionCondition.java]: @asolntsev Maybe it's will be better to import `org.checkerframework.checker.nullness.qual.Nullable;` to match implemented method signature from guava library?

File: src/main/java/com/codeborne/selenide/impl/FilteringCollection.java [hunk 2]
```diff
@@ -3,9 +3,9 @@
 import com.codeborne.selenide.Condition;
 import com.codeborne.selenide.Driver;
 import com.google.common.base.Predicate;
-import org.checkerframework.checker.nullness.compatqual.NullableDecl;
 import org.openqa.selenium.WebElement;
 
+import javax.annotation.Nullable;
```
Comment (rosolko) [inline src/main/java/com/codeborne/selenide/impl/FilteringCollection.java]: Same here

**Comment by martinschneider** [comment]
What's the status of this? Any help required? I would like to use Selenide with Selenium 4.0.0-alpha3. Let me know if there's a way to contribute to the upgrade efforts.

**Comment by asolntsev** [comment]
@martinschneider We are not working on this pull request. Since Selenium 4.0 is not released yet, I don't see any reason why we should upgrade to it. If I understood correctly, it will not be released soon. 

By the way, why do you want to use Selenium 4.0.0-alpha3?

**Comment by martinschneider** [comment]
@asolntsev Fair enough if the plan is to wait for an official release. My point is, that once Selenium 4 is out, people will start upgrading and expect other libraries (like Selenide) to "just work".

While I agree that it doesn't make sense, to _release_ a version of Selenide which uses an alpha version of Selenium, I think it would make a lot of sense to keep this PR active and track the implications and amount of work supporting Selenium 4 will bring.

After all, that's what alpha releases are there for, to give tool developers like us a heads-up of what's coming.

**Comment by asolntsev** [comment]
@martinschneider Thank you for the remainder!
I am absolutely agree, alpha releases are there for us to try them as early as possible.

I have merged master to this PR, fixed failing tests. Will keep it up-to-date until Selenium 4 is released.

**Comment by rosolko** [comment]
@martinschneider Hey, but the way, if you want to use version from this branch you can easily build it and get for using via jitpack:
https://jitpack.io/#selenide/selenide/upgrade-to-selenium4-SNAPSHOT

**Comment by asolntsev** [comment]
Overwritten by https://github.com/selenide/selenide/pull/1099

**Pull request closed without merging**